sdk/core/core/inc/az_http_transport.h

```diff
@@ -85,27 +85,88 @@ AZ_NODISCARD az_result
+ *         - #AZ_OK if successful
+ */
+AZ_NODISCARD az_result
+az_http_request_get_method(_az_http_request const* request, az_http_method* out_method);
+
+/**
+ * @brief Get url from an HTTP request.
+ *
+ * @remarks This function is expected to be used by transport layer only.
+ *
+ * @param[in] request The HTTP request from which to get the url.
+ * @param[out] out_url Pointer to write the HTTP URL to.
+ *
+ * @retval An #az_result value indicating the result of the operation:
+ *         - #AZ_OK if successful
  */
-AZ_NODISCARD az_result az_http_request_get_parts(
-    _az_http_request const* request,
-    az_http_method* out_method,
-    az_span* out_url,
-    az_span* out_body);
+AZ_NODISCARD az_result az_http_request_get_url(_az_http_request const* request, az_span* out_url);
+
+/**
+ * @brief Get body from an HTTP request.
+ *
+ * @remarks This function is expected to be used by transport layer only.
+ *
+ * @param[in] request The HTTP request  from which to get the body.
+ * @param[out] out_body Pointer to write the HTTP request body to.
+ *
+ * @retval An #az_result value indicating the result of the operation:
+ *         - #AZ_OK if successful
+ */
+AZ_NODISCARD az_result az_http_request_get_body(_az_http_request const* request, az_span* out_body);
+
+/**
+ * @brief This function is expected to be used by transport adapters like curl. Use it to write
+ * content from \p source to \p response.
+ *
+ * @remarks The \p source can be an empty #az_span. If so, nothing will be written.
+ *
+ * @param[in] response Pointer to an az_http_response.
+ * @param[in] source This is an az_span with the content to be written into response.
+ * @return An #az_result value indicating the result of the operation:
+ *         - #AZ_OK if successful
+ *         - #AZ_ERROR_INSUFFICIENT_SPAN_SIZE if the \p response buffer is not big enough to contain
+ * the \p source content
+ */
+AZ_NODISCARD az_result az_http_response_write_span(az_http_response* response, az_span source);
+
+/**
+ * @brief Returns the number of headers within the request.
+ * Each header is an #az_pair.
+ *
+ * @param[in] request Pointer to an az_http_request to be used by this function.
+ * @return Number of headers in the request.
+ */
+AZ_NODISCARD int32_t az_http_request_headers_count(_az_http_request const* request);
+
+/**
+ * @brief Send an HTTP request through the wire and write the response into \p p_response.
+ *
+ * @param[in] p_request Points to an az_http_request that contains the settings and data that is
+ * used to send the request through the wire.
+ * @param[out] p_response Points to an az_http_response where the response from the wire will be
+ * written.
+ * @return An #az_result value indicating the result of the operation:
+ *         - #AZ_OK if successful
+ *         - #CURLE_WRITE_ERROR if there was any issue while trying to write into \p response. It
+ * might means that there was not enough size on \p response to hold the entire response from
+ * network.
```

Comment:
I don't think `CURLE_WRITE_ERROR` is an `az_result`. Instead document what `az_result` value we return in these cases.

```suggestion
 *         - #AZ_ERROR_HTTP_RESPONSE_OVERFLOW if there was any issue while trying to write into \p p_response. It
 * might mean that there was not enough space in \p p_response to hold the entire response from
 * the network.
```